### DISCUSSION TITLE
Upcoming changes?

### DISCUSSION THREAD

**Discussion by johr** [docToolchain/docToolchain#567] [Q&A] [answered]
I recently used doctoolchain. Really like it! 
If I remember correctly, I read something about refactoring, ... going on in the project. 
Is this right? If so what would be the changes?
Thanks. BR, johr

**Comment by rdmueller** [comment, marked as answer]
Great that you ask!
In the original design, docToolchain was part of your gradle build (if you used gradle) or a second build.
@jakubjab soon refactored it in such a way that you could also use it as command line tool.

And since many devs where scared by gradle because they use build tool X, Y or Z, I thought it was a good idea to hide the technologies behind docToolchain.

The new version (which I started to document here: https://doctoolchain.github.io/docToolchain/v2.0.x/#_how_to_install_doctoolchain) now is a wrapper like the gradle wrapper. 
This wrapper check if docToolchain is already installed, if docker is available etc. and is able to install docToolchain as CLI or use it as docker image.

This decouples your repository from docToolchain. You will only need to have the wrapper and a config in your repository. Nothing else.

A second major change is that we've now added jBake as Static Site Generator and the ability to download and install documentation templates.
And again, docToolchain tries to hide the technology and boiler plate code from your project.
jBake will not install a theme or assets to your repository but there are mechanisms to change the theme etc.

The new version is already in a state where I start to use it in projects in order to test it. Feel free to also give it a try and report problems!

If you want to know more about the new version...

* check out the ng-branch
* here is an english video of my talk @ FOSDEM: https://fiveandahalfstars.ninja/talks/details/doctoolchain
* and here is a german video of my talk about jBake: https://fiveandahalfstars.ninja/talks/details/doku-microsites-mit-jbake

**Reply by johr** [reply]
thanks for the detailed response. sounds good. I will give it a try!
